tests/integration/test_multiple_response.py

```diff
@@ -664,3 +665,72 @@ def test_mr_x_num_cols_margin_pruned(self):
         )
         for actual in margin:
             np.testing.assert_array_equal(actual, expected)
+
+    def test_num_x_mr_props_by_row(self):
```

Comment:
I'd really rather see a unit test rather than an integration test for this fix.  Perhaps you can add some unit testing in this area and remove this integration test.